pytensor/tensor/subtensor.py

```diff
@@ -2486,15 +2532,104 @@ def gen_num(typen):
         return code
 
     def c_code(self, node, name, input_names, output_names, sub):
-        if numpy_version < "1.8.0" or using_numpy_2:
-            raise NotImplementedError
-
         x, y, idx = input_names
-        out = output_names[0]
+        [out] = output_names
         copy_of_x = self.copy_of_x(x)
         params = sub["params"]
         fail = sub["fail"]
 
+        x_, y_, idx_ = node.inputs
+        y_dtype = y_.type.dtype_specs()[1]
+        idx_dtype = idx_.type.dtype_specs()[1]
+        out_dtype = node.outputs[0].type.dtype_specs()[1]
+        y_bcast = y_.type.broadcastable != idx_.type.broadcastable
+        if (
+            x_.type.ndim == 1
+            and x_.type.dtype not in complex_dtypes
+            and not y_bcast
+            and y_.type.dtype not in complex_dtypes
+        ):
+            # Simple implementation for vector x, y cases
```

Comment:
You don't actually check that `y_` is ndim == 1, is it because no broadcast is going on so it's not necessary?